tests/unit/Exception/HubspotExceptionTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+declare(strict_types=1);
+
+namespace SevenShores\Hubspot\Tests\unit\Exception;
+
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Exception\ServerException;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Response;
+use PHPUnit\Framework\TestCase;
+use SevenShores\Hubspot\Exceptions\BadRequest;
+use SevenShores\Hubspot\Exceptions\HubspotException;
+use function GuzzleHttp\Psr7\stream_for;
+
+class HubspotExceptionTest extends TestCase
+{
+    const EXAMPLE_TOKEN = '8907e60c-600d-4af8-a987-191c104a215c';
+
+    public function testCreateExceptionFromGuzzleRequestException()
+    {
+        $e = new RequestException('Request Failed', new Request('GET', 'https://api.hubspot.com/x'));
+
+        $hubspotException = HubspotException::create($e);
+
+        $this->assertInstanceOf(HubspotException::class, $hubspotException);
+        $this->assertNull($hubspotException->getResponse());
+    }
+
+    public function testCreateExceptionFromGuzzleClientException()
+    {
+        $e = ClientException::create(
+            new Request(
+                'GET',
+                sprintf('https://api.hubapi.com/deals/v1/deal/12345?access_token=%s', self::EXAMPLE_TOKEN)
+            ),
+            new Response(
+                400,
+                [],
+                stream_for('{"status":"error","message":"xyz"}')
+            )
+        );
+
+        $hubspotException = BadRequest::create($e);
+
+        $this->assertInstanceOf(BadRequest::class, $hubspotException);
+        $this->assertNotContains(self::EXAMPLE_TOKEN, $hubspotException->getMessage());
+        $this->assertSame($e->getResponse(), $hubspotException->getResponse());
+        $this->assertSame('Client error: `GET https://api.hubapi.com/deals/v1/deal/12345?access_token=***` resulted in a `400 Bad Request` response:
+{"status":"error","message":"xyz"}
+', $hubspotException->getMessage());
+    }
+
+    public function testCreateExceptionFromGuzzleServerException()
+    {
+        $e = ServerException::create(
+            new Request(
+                'GET',
+                sprintf('https://api.hubapi.com/deals/v1/deal/12345?hapikey=%s', self::EXAMPLE_TOKEN)
+            ),
+            new Response(
+                502,
+                [],
+                stream_for('<!DOCTYPE html>
+<!--[if lt IE 7]> <html class="no-js ie6 oldie" lang="en-US"> <![endif]-->')
+            )
+        );
+
+        $hubspotException = HubspotException::create($e);
+
+        $this->assertInstanceOf(HubspotException::class, $hubspotException);
+        $this->assertNotContains(self::EXAMPLE_TOKEN, $hubspotException->getMessage());
```

Comment:
Could You change self to static at this place?
I understand that won’t  be redefined but it is our style
over time we will change all private to protected and self to static (where it is necessary)